### ISSUE TITLE
kubectl have nil pointer panic

### ISSUE THREAD

**Issue by mingzhu02** [kubernetes/kubernetes#76412]
kubectl version 1.9.6
```
# kubectl -s 10.56.64.34  delete pod obj-detect-fast-yz-deploy-8876d7688-fh5xf
panic: runtime error: invalid memory address or nil pointer dereference
[signal SIGSEGV: segmentation violation code=0x1 addr=0x58 pc=0x15b8633]

goroutine 1 [running]:
k8s.io/kubernetes/pkg/kubectl.ReaperFor(0x0, 0x0, 0x176ff37, 0x3, 0x0, 0x0, 0x0, 0xc420a034a0, 0x7f6002, 0xbf23cbb40809b034)
        /workspace/anago-v1.9.6-beta.0.17+9f8ebd171479be/src/k8s.io/kubernetes/_output/dockerized/go/src/k8s.io/kubernetes/pkg/kubectl/delete.go:82 +0x1373
k8s.io/kubernetes/pkg/kubectl/cmd/util.(*ring1Factory).Reaper(0xc4208d0f90, 0xc4201842a0, 0x2a64900, 0x1a0020644000, 0xb8, 0xb8)
        /workspace/anago-v1.9.6-beta.0.17+9f8ebd171479be/src/k8s.io/kubernetes/_output/dockerized/go/src/k8s.io/kubernetes/pkg/kubectl/cmd/util/factory_object_mapping.go:295 +0x151
k8s.io/kubernetes/pkg/kubectl/cmd/util.(*factory).Reaper(0xc4208d0fc0, 0xc4201842a0, 0xc420644000, 0x7f53e92066c8, 0x0, 0x1a77de0)
        <autogenerated>:1 +0x47
k8s.io/kubernetes/pkg/kubectl/cmd.ReapResult.func1(0xc4201843f0, 0x0, 0x0, 0x2c13980, 0x1c52e3e)
        /workspace/anago-v1.9.6-beta.0.17+9f8ebd171479be/src/k8s.io/kubernetes/_output/dockerized/go/src/k8s.io/kubernetes/pkg/kubectl/cmd/delete.go:250 +0xe9
k8s.io/kubernetes/pkg/kubectl/resource.ContinueOnErrorVisitor.Visit.func1(0xc4201843f0, 0x0, 0x0, 0x0, 0x0)
        /workspace/anago-v1.9.6-beta.0.17+9f8ebd171479be/src/k8s.io/kubernetes/_output/dockerized/go/src/k8s.io/kubernetes/pkg/kubectl/resource/visitor.go:396 +0x164
k8s.io/kubernetes/pkg/kubectl/resource.DecoratedVisitor.Visit.func1(0xc4201843f0, 0x0, 0x0, 0x7, 0xa)
        /workspace/anago-v1.9.6-beta.0.17+9f8ebd171479be/src/k8s.io/kubernetes/_output/dockerized/go/src/k8s.io/kubernetes/pkg/kubectl/resource/visitor.go:372 +0xe7
k8s.io/kubernetes/pkg/kubectl/resource.FlattenListVisitor.Visit.func1(0xc4201843f0, 0x0, 0x0, 0xc420b5c090, 0x7f53e8ffdb80)
        /workspace/anago-v1.9.6-beta.0.17+9f8ebd171479be/src/k8s.io/kubernetes/_output/dockerized/go/src/k8s.io/kubernetes/pkg/kubectl/resource/visitor.go:433 +0x4fe
k8s.io/kubernetes/pkg/kubectl/resource.(*Info).Visit(0xc4201843f0, 0xc420b5c090, 0x0, 0x4109e7)
        /workspace/anago-v1.9.6-beta.0.17+9f8ebd171479be/src/k8s.io/kubernetes/_output/dockerized/go/src/k8s.io/kubernetes/pkg/kubectl/resource/visitor.go:105 +0x42
k8s.io/kubernetes/pkg/kubectl/resource.VisitorList.Visit(0xc4205ded80, 0x1, 0x1, 0xc420b5c090, 0x1, 0xc420b5c090)
        /workspace/anago-v1.9.6-beta.0.17+9f8ebd171479be/src/k8s.io/kubernetes/_output/dockerized/go/src/k8s.io/kubernetes/pkg/kubectl/resource/visitor.go:234 +0x63
k8s.io/kubernetes/pkg/kubectl/resource.(*VisitorList).Visit(0xc4203cf460, 0xc420b5c090, 0x7f53e92066c8, 0x0)
        <autogenerated>:1 +0x58
k8s.io/kubernetes/pkg/kubectl/resource.FlattenListVisitor.Visit(0x2a58400, 0xc4203cf460, 0xc4205dac40, 0xc4205db0c0, 0x60000000001, 0xc4205db0c0)
        /workspace/anago-v1.9.6-beta.0.17+9f8ebd171479be/src/k8s.io/kubernetes/_output/dockerized/go/src/k8s.io/kubernetes/pkg/kubectl/resource/visitor.go:428 +0x9e
k8s.io/kubernetes/pkg/kubectl/resource.(*FlattenListVisitor).Visit(0xc4203cf480, 0xc4205db0c0, 0x28, 0xc420644000)
        <autogenerated>:1 +0x58
k8s.io/kubernetes/pkg/kubectl/resource.DecoratedVisitor.Visit(0x2a58380, 0xc4203cf480, 0xc4205ded90, 0x2, 0x2, 0xc4203cf4e0, 0xc4205ded01, 0xc4203cf4e0)
        /workspace/anago-v1.9.6-beta.0.17+9f8ebd171479be/src/k8s.io/kubernetes/_output/dockerized/go/src/k8s.io/kubernetes/pkg/kubectl/resource/visitor.go:363 +0x9b
k8s.io/kubernetes/pkg/kubectl/resource.(*DecoratedVisitor).Visit(0xc420b5c060, 0xc4203cf4e0, 0xc4205deda0, 0xc420a03b60)
        <autogenerated>:1 +0x62
k8s.io/kubernetes/pkg/kubectl/resource.ContinueOnErrorVisitor.Visit(0x2a58280, 0xc420b5c060, 0xc420184460, 0x0, 0x2)
        /workspace/anago-v1.9.6-beta.0.17+9f8ebd171479be/src/k8s.io/kubernetes/_output/dockerized/go/src/k8s.io/kubernetes/pkg/kubectl/resource/visitor.go:391 +0xe4
k8s.io/kubernetes/pkg/kubectl/resource.(*ContinueOnErrorVisitor).Visit(0xc4205deda0, 0xc420184460, 0x411148, 0x70)
        <autogenerated>:1 +0x4f
k8s.io/kubernetes/pkg/kubectl/resource.(*Result).Visit(0xc42042a300, 0xc420184460, 0xc420add7c0, 0x2a77880)
        /workspace/anago-v1.9.6-beta.0.17+9f8ebd171479be/src/k8s.io/kubernetes/_output/dockerized/go/src/k8s.io/kubernetes/pkg/kubectl/resource/result.go:98 +0x62
k8s.io/kubernetes/pkg/kubectl/cmd.ReapResult(0xc42042a300, 0x2a84680, 0xc4208d0fc0, 0x2a57480, 0xc42000c018, 0xc420500001, 0x0, 0xffffffffffffffff, 0xc4203b0000, 0x2a77880, ...)
        /workspace/anago-v1.9.6-beta.0.17+9f8ebd171479be/src/k8s.io/kubernetes/_output/dockerized/go/src/k8s.io/kubernetes/pkg/kubectl/cmd/delete.go:245 +0x16c
k8s.io/kubernetes/pkg/kubectl/cmd.(*DeleteOptions).RunDelete(0xc42087e0b0, 0xc4203e8240, 0x0)
        /workspace/anago-v1.9.6-beta.0.17+9f8ebd171479be/src/k8s.io/kubernetes/_output/dockerized/go/src/k8s.io/kubernetes/pkg/kubectl/cmd/delete.go:235 +0xd2
k8s.io/kubernetes/pkg/kubectl/cmd.NewCmdDelete.func1(0xc4203e8240, 0xc4205073c0, 0x2, 0x4)
        /workspace/anago-v1.9.6-beta.0.17+9f8ebd171479be/src/k8s.io/kubernetes/_output/dockerized/go/src/k8s.io/kubernetes/pkg/kubectl/cmd/delete.go:142 +0x178
k8s.io/kubernetes/vendor/github.com/spf13/cobra.(*Command).execute(0xc4203e8240, 0xc420507380, 0x4, 0x4, 0xc4203e8240, 0xc420507380)
        /workspace/anago-v1.9.6-beta.0.17+9f8ebd171479be/src/k8s.io/kubernetes/_output/dockerized/go/src/k8s.io/kubernetes/vendor/github.com/spf13/cobra/command.go:603 +0x234
k8s.io/kubernetes/vendor/github.com/spf13/cobra.(*Command).ExecuteC(0xc4203a2240, 0x8000101, 0x0, 0xffffffffffffffff)
        /workspace/anago-v1.9.6-beta.0.17+9f8ebd171479be/src/k8s.io/kubernetes/_output/dockerized/go/src/k8s.io/kubernetes/vendor/github.com/spf13/cobra/command.go:689 +0x2fe
k8s.io/kubernetes/vendor/github.com/spf13/cobra.(*Command).Execute(0xc4203a2240, 0xc4208d0fc0, 0x2a57440)
        /workspace/anago-v1.9.6-beta.0.17+9f8ebd171479be/src/k8s.io/kubernetes/_output/dockerized/go/src/k8s.io/kubernetes/vendor/github.com/spf13/cobra/command.go:648 +0x2b
k8s.io/kubernetes/cmd/kubectl/app.Run(0x0, 0x0)
        /workspace/anago-v1.9.6-beta.0.17+9f8ebd171479be/src/k8s.io/kubernetes/_output/dockerized/go/src/k8s.io/kubernetes/cmd/kubectl/app/kubectl.go:41 +0xd5
main.main()
        /workspace/anago-v1.9.6-beta.0.17+9f8ebd171479be/src/k8s.io/kubernetes/_output/dockerized/go/src/k8s.io/kubernetes/cmd/kubectl/kubectl.go:27 +0x26
```

**Comment by Pingan2017** [comment]
what the version of your server? 
using kubectl version?

**Comment by mingzhu02** [comment]
> what the version of your server?
> using kubectl version?

kube-apiserver version is 1.9.6

**Comment by tanshanshan** [comment]
the version you used is old, 
Even it is a bug , it can not be fixed in 1.9.6
you should better upgrade your version.

**Comment by mingzhu02** [comment]
i try  latest version,the bug has been fixed,thanks!
